python/tank/deploy/tank_commands/validate_config.py

```diff
@@ -46,36 +103,112 @@ def run_interactive(self, log, args):
         :param log: std python logger
         :param args: command line args
         """
-        if len(args) != 0:
-            raise TankError("This command takes no arguments!")
-        return self._run(log)
+
+        self._interactive = True
+
+        parser = argparse.ArgumentParser(
+            prog="./tank validate",
+            description="Environment validation command."
+        )
+
+        # a list of environment names
+        parser.add_argument(
+            "envs",
+            metavar="<env_name>",
+            type=str,
+            nargs="*",
+            help=self.parameters["envs"]["description"],
+        )
+
+        # mutually exclusive dump types, full and sparse
+        dump_type_group = parser.add_mutually_exclusive_group()
+        dump_type_group.add_argument(
+            "-f", "--dump",
+            action="store_true",
+            help=self.parameters["dump"]["description"],
+        )
+        dump_type_group.add_argument(
+            "-s", "--dump-sparse",
+            action="store_true",
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # the output directory for the modified environment files
+        parser.add_argument(
+            "-l", "--dump-location",
+            metavar="<output_directory>",
+            type=str,
+            help=self.parameters["dump_sparse"]["description"],
+        )
+
+        # turn on debug comments in the dumped environments
+        parser.add_argument(
+            "--dump-debug",
+            action="store_true",
+            help=self.parameters["dump_debug"]["description"],
+        )
+
+        # calling vars() on the returned namespace object essentially
+        # translates it into a dictionary. In addition, argparse naturally
+        # translates the flags we used to the matching parameters that
+        # are defined by the action (ex: "--dump-debug" becomes "dump_debug").
+        # so we're left with a dict that should look just like the params
+        # supplied to the non-interactive version of this action.
+        parsed_args = vars(parser.parse_args(args))
+
+        # validate the parsed args just like the non-interactive params.
+        computed_params = self.__validate_parameters(parsed_args)
```

Comment:
The args are part of a mutually exclusive group, so the command will error out if both are supplied. https://github.com/shotgunsoftware/tk-core/blob/35511_dump_sparse/python/tank/deploy/tank_commands/validate_config.py#L123-L134

Here's the current output if both are supplied:

```
> ./tank validate shot_step --dump --dump-sparse
[...]
usage: ./tank validate [-h] [-f | -s] [-l <output_directory>] [--dump-debug]
                       [<env_name> [<env_name> ...]]
./tank validate: error: argument -s/--dump-sparse: not allowed with argument -f/--dump
```

Let me know if that doesn't seem reasonable.

I guess the validation currently wouldn't handle the non-interactive case though. I will address that part.